shiny/ui/_html_deps_external.py

```diff
@@ -26,24 +26,50 @@
 ThemeProvider = Union[Tagifiable, HTMLDependency, List[HTMLDependency]]
 
 
+def mark_shiny_theme_stylesheets(dep: HTMLDependency, name: str = "") -> HTMLDependency:
+    if not isinstance(dep, HTMLDependency):
+        raise ValueError(
+            "Expected an HTMLDependency, but received a value with type "
+            f"{type(dep)}."
+        )
+
+    if len(dep.stylesheet) > 0:
+        for sheet in dep.stylesheet:
+            sheet["data-shiny-theme"] = name  # type: ignore
```

Comment:
Does this data attribute logic have to come from Shiny? Seems like shinyswatch should be responsible for this?

Reply:
How would shinyswatch know which dependencies were provided to the `theme` argument?

Reply:
The immediate goal is to enable the shinyswatch theme-picker to find the initial theme of the app. It's trivial (and actually unnecessary) to do this when the theme comes from shinyswatch, but we need some way to identify the stylesheets from the page-level `theme`.

Reply:
Btw, you can see the feature this enables in action in the PR description of https://github.com/posit-dev/py-shinyswatch/pull/39